include/swift/AST/Types.h

```diff
@@ -3917,6 +3917,24 @@ namespace Lowering {
   class TypeConverter;
 };
 
+class SILUncommonInfo {
```

Comment:
This is a very generic name for such a specific type.  Can it be made a nested type of `SILFunctionType`?

Reply:
I tried that earlier, but the template argument in `SILFunctionType`'s `TrailingObjects` needs the type to be complete. I suspect `SILParameterInfo` and `SILResultInfo` have been defined separately instead of as nested inside for a similar reason.

Maybe I should rename it to `SILCallingConvUncommonInfo` or something else which is more precise?

Reply:
Let me merge this for now, since I don't want to rerun coordinated testing with LLDB again, and this makes doing follow-up work easier. I will change the name in a new PR which can be Swift-only.

Reply:
Okay.  `SILFunctionTypeUncommonInfo` seems reasonable.